cli/monitors/terminal_outputter.py

```diff
@@ -0,0 +1,770 @@
+                        PP.println()
+                PP.println("         |_  {}... "
+                           .format(SimplePrinter.format_state_event(step.start_event)))
+            else:
+                PP.print("[init] Executing state {}... ".format(step.name))
+        if step.skipped:
+            PP.println("skipped")
+        else:
+            self.current_step.append({'endl': True})
+
+    def step_state_minion_finished(self, step, minion):
+        if not step.targets[minion]['success']:
+            if step.name not in self.errors:
+                self.errors[step.name] = OrderedDict()
+            self.errors[step.name][minion] = step.targets[minion]['event']
+
+        if step.order > 0:
+            PP.print("         {}... ".format(minion))
+        else:
+            PP.print("               in {}... ".format(minion))
+        if step.targets[minion]['success']:
+            PP.println("ok")
+        else:
+            PP.println("fail")
+
+    def step_state_finished(self, step):
+        self.current_step.pop()
+
+    def step_state_result(self, step, event):
+        if self.current_step:
+            if not self.current_step[-1]['endl']:
+                self.current_step[-1]['endl'] = True
+                PP.println()
+        if event.name == event.state_id:
+            PP.print("         |_  {}: {} ".format(event.minion, event.name))
+        else:
+            PP.print("         |_  {}: {}({}) ".format(event.minion, event.state_id, event.name))
+
+        if event.result:
+            PP.println("ok")
+        else:
+            PP.println("fail")
+
+    @staticmethod
+    def format_runner_event(event):
+        if event.fun.startswith('runner.'):
+            fun_name = event.fun[7:]
+        else:
+            fun_name = event.fun
+        args = ""
+        first = True
+        for arg in event.args:
+            if isinstance(arg, dict):
+                for key, val in arg.items():
+                    if first:
+                        args += "{}={}".format(key, val)
+                        first = False
+                    else:
+                        args += ", {}={}".format(key, val)
+            else:
+                if first:
+                    first = True
+                    args += "{}".format(arg)
+                else:
+                    args += ", {}".format(arg)
+        return "{}({})".format(fun_name, args)
+
+    @staticmethod
+    def format_state_event(event):
+        fun_name = event.fun
+        args = ""
+        first = True
+        for arg in event.args:
+            if isinstance(arg, dict):
+                for key, val in arg.items():
+                    if first:
+                        args += "{}={}".format(key, val)
+                        first = False
+                    else:
+                        args += ", {}={}".format(key, val)
+            else:
+                if first:
+                    first = True
```

Comment:
As above -- `if first: first = True` seems to be setting first to what it's already set to.